readthedocs/donate/models.py

```diff
@@ -104,6 +105,14 @@ def incr(self, type):
         # TODO: Support redis, more info on this PR
         # github.com/rtfd/readthedocs.org/pull/2105/files/1b5f8568ae0a7760f7247149bcff481efc000f32#r58253051
 
+    def incr_project(self, type, project):
+        """Add to the number of times this action has been performed, stored in the DB"""
+        assert type in IMPRESSION_TYPES
+        day = get_ad_day()
+        impression, _ = self.project_impressions.get_or_create(date=day, project=project)
+        setattr(impression, type, models.F(type) + 1)
+        impression.save()
+
```

Comment:
To reduce duplication, this could be combined with the method above, with an optional keyword arg `project`